upgrade/sql/8.1.0.sql

```diff
@@ -58,6 +58,13 @@ ALTER TABLE `PREFIX_product_shop` MODIFY COLUMN `redirect_type` ENUM(
 UPDATE `PREFIX_product` SET `redirect_type` = 'default' WHERE `redirect_type` = '404' OR `redirect_type` = '' OR `redirect_type` IS NULL;
 UPDATE `PREFIX_product_shop` SET `redirect_type` = 'default' WHERE `redirect_type` = '404' OR `redirect_type` = '' OR `redirect_type` IS NULL;
 
+ALTER TABLE `PREFIX_product` MODIFY COLUMN `redirect_type` ENUM(
+   '404','410','301-product','302-product','301-category','302-category','200-displayed','404-displayed','410-displayed','default'
```

Comment:
Already present few lines above, you can modify the original one ;-)

Reply:
Thanks for the review :) I think we need to do this in 3 steps:
- Add the ENUMS while keeping the `''` so as not to lose the existing values.
- Update to `default` where `''` and ...
- Remove the unnecessary `''`

Reply:
Yes, exactly as you say. :-)

I kept the empty string in the enum, because products created pre 1.7.8.0 had it by default - https://github.com/PrestaShop/PrestaShop/blob/aab3f53c9c31d79a6c60d9744d2a175c4d6c5194/install-dev/data/db_structure.sql#L1647

And also, it was implemented in 8.1.0, so allowing that empty string was to avoid a BC break.

Reply:
Soooo, actually, I think we should do this in 9.0.0 and keep this like it is. 🤔

Reply:
Ok, well seen !

I changed the updating of data and columns in the v9 script, this way we catch up with potential insertions with empty string